src/main/java/seedu/findvisor/logic/commands/RescheduleCommand.java

```diff
@@ -0,0 +1,233 @@
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_START_DATETIME + "START DATETIME] "
+            + "[" + PREFIX_END_DATETIME + "END DATETIME] "
+            + "[" + PREFIX_MEETING_REMARK + "REMARK]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_END_DATETIME + "22-02-2024T16:00 "
+            + PREFIX_MEETING_REMARK + "Extended by one hour";
+
+    public static final String MESSAGE_RESCHEDULE_SUCCESS = "Rescheduled meeting with %1$s.\n%2$s";
+    public static final String MESSAGE_CANNOT_RESCHEDULE_NON_EXISTENT_MEETING = "To reschedule a meeting"
+            + " you must first schedule a meeting with the contact!";
+
+    private final Index targetIndex;
+    private final EditMeetingDescriptor editMeetingDescriptor;
+
+    /**
+     * Creates an RescheduleCommand to reschedule a meeting with the specified {@code Person}
+     */
+    public RescheduleCommand(Index targetIndex, EditMeetingDescriptor editMeetingDescriptor) {
+        requireNonNull(editMeetingDescriptor);
+        this.targetIndex = targetIndex;
+        this.editMeetingDescriptor = editMeetingDescriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(targetIndex.getZeroBased());
+        if (personToEdit.getMeeting().isEmpty()) {
+            throw new CommandException(MESSAGE_CANNOT_RESCHEDULE_NON_EXISTENT_MEETING);
+        }
+        Person editedPerson = createEditedPerson(personToEdit, editMeetingDescriptor);
+
+        if (editMeetingDescriptor.getStart().isPresent()
+                && !DateTimeUtil.isAfterCurrentDateTime(editMeetingDescriptor.getStart().get())) {
+            throw new CommandException(MESSAGE_CANNOT_SCHEDULE_MEETING_IN_THE_PAST);
+        }
+
+        model.setPerson(personToEdit, editedPerson);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+        return new CommandResult(String.format(MESSAGE_RESCHEDULE_SUCCESS,
+                editedPerson.getName(),
+                Messages.format(editedPerson.getMeeting().get())));
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the same details of {@code personToEdit}
+     * with the meeting details edited.
+     *
+     * @param personToEdit Person to edit. Must have a meeting scheduled.
+     * @param editMeetingDescriptor Details to edit the meeting with.
+     * @return Person with the meeting details edited.
+     */
+    private static Person createEditedPerson(Person personToEdit, EditMeetingDescriptor editMeetingDescriptor) {
+        assert personToEdit != null && editMeetingDescriptor != null;
```

Comment:
Would it be better to use `requireAllNonNull()` instead here?

Reply:
`requireAllNonNull()` throws `NullPointerException`, I think assertion will be more appropriate here since it should not be possible for the user or environment to trigger this, if we reach this point in execution with either of the parameters as `null` then its a programmer problem

> - The raising of an exception indicates an unusual condition created by the user (e.g. user inputs an unacceptable input) or the environment (e.g., a file needed for the program is missing).
> - An assertion failure indicates the programmer made a mistake in the code (e.g., a null value is returned from a method that is not supposed to return null under any circumstances).

Reply:
Thanks for clarifying! @Dethada